### ISSUE TITLE
prefer-template fix introducing extra space

### ISSUE THREAD

**Issue by maloguertin** [prettier/eslint-plugin-prettier#153]
**What version of `eslint` are you using?**
`"eslint": "^5.12.0"`

**What version of `prettier` are you using?**
`"prettier": "^1.16.1"`

**What version of `eslint-plugin-prettier` are you using?**
`"eslint-config-prettier": "^3.3.0"` 

**Please paste any applicable config files that you're using (e.g. `.prettierrc` or `.eslintrc` files)**
```
{
  "parser": "babel-eslint",
  "plugins": ["prettier", "react", "jest", "cypress"],
  "env": {
    "es6": true,
    "browser": true,
    "node": true,
    "jasmine": true,
    "jest/globals": true
  },
  "parserOptions": {
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "settings": {
    "import/resolver": {
      "webpack": {
        "config": "webpack.common.js"
      }
    }
  },
  "rules": {
    "arrow-body-style": "off",
    "no-unused-vars": [
      "error",
      { "vars": "all", "args": "after-used", "ignoreRestSiblings": true }
    ],
    "react/jsx-uses-react": "error",
    "react/jsx-uses-vars": "error",
    "react/prop-types": 2,
    "import/no-extraneous-dependencies": [
      "error",
      { "devDependencies": ["**/*.test.js", "**/setupTests.js"] }
    ],
    "max-len": 0,
    "react/jsx-filename-extension": 0,
    "no-restricted-syntax": ["error", "LabeledStatement", "WithStatement"],
    "cypress/globals": true
  },
  "extends": ["airbnb"]
}
```
```
{
  "trailingComma": "es5",
  "tabWidth": 2,
  "semi": false,
  "singleQuote": true
}
```

**What source code are you linting?**
```js
const d = 'd'
const abcd = 'abc' + d
```

**What did you expect to happen?**
```
const d = 'd'
const abcd = `abc${d}`
```

**What actually happened?**
```
const d = 'd'
const abcd = `abc${ d}`
```

basically I think the package disables the linting rule that allows for the intermediate `abc${ d}` to be changed to `abc${d}`

**Comment by BPScott** [comment]
[eslint-config-prettier disables template-curly-spacing](https://github.com/prettier/eslint-config-prettier/blob/0f131661f718e278ec1e51529f56450a61f10b11/index.js#L82). The rational for this is that you should trust how prettier formats your code instead instead of enabling other stylistic rules - otherwise you end up getting into a fight where prettier things code should be formatted one way, and some other rule thinks otherwise.

Though I must admit I am slightly confused by this as I see no mention of enabling the prettier/prettier rule, or extending from eslint-config-prettier as per the [recommended configuration](https://github.com/prettier/eslint-plugin-prettier#recommended-configuration) so I'm not sure if you're even enabling and correctly configuring eslint-plugin-prettier in the first place.

**Comment by j-f1** [comment]
Unless you need to customize Prettier specifically for ESLint (which you shouldn’t in most cases — use `.prettierrc` instead), you just need to install `eslint-plugin-prettier` and extend `plugin:prettier/recommended`, which enables `eslint-config-prettier` and the `prettier/prettier` rule that runs Prettier on your code.

**Comment by maloguertin** [comment]
The problem here is that prettier runs first with prettier-eslint, therefore it won't fix the whitespace introduced by eslint unless you run it with `--prettier-last`

**Comment by j-f1** [comment]
@maloguertin `prettier-eslint !== eslint-plugin-prettier`. `eslint-plugin-prettier` runs Prettier inside of ESLint.

**Comment by maloguertin** [comment]
@j-f1 oops, brain fart, feel free to close.
